p4-16/spec/P4-16-draft-spec.mdk

```diff
@@ -4876,22 +4877,18 @@ match_kind {
 ~ End P4Example
 
 These identifiers correspond to the P4~14~ match kinds with the same
-names. The semantics of these annotations is actually irrelevant for
-describing the behavior of the P4 abstract machine; how they are used
-influences only the control-plane API and the actual implementation of
-the look-up table. From the point of view of the P4 program, a look-up
-table is an abstract finite map that is given a key and produces as a
-result either an action or a "miss" indication, as described in
-Section [#sec-mau-semantics].
```

Comment:
The semantics is that given a key you get back an action. How that is done is irrelevant.

I can use two tables exactly like t1, if I populate them with different entries I will get different behaviors. That's what you are doing in your example. But both tables take a key and return an action - that is the dataplane semantics.